### PR TITLE
adding devtest labs facts

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
Adding devtest lab facts module

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
azure_rm_devtestlab

### CODE REVIEW

**Comment by Fred-sun** [comment]
@yungezz @yuwzho review_need

File: lib/ansible/modules/cloud/azure/azure_rm_devtestlab_facts.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+                - Lab provisioning state.
+            returned: always
+            type: str
+            sample: Succeeded
+        artifacts_storage_account:
+            description:
+                - Artifacts storage account ID.
+            returned: always
+            type: str
+            sample: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/testrg/providers/Microsoft.Storage/storageAccounts/testlab6346
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_devtestlab_facts.py:109]: sample name consistent

File: lib/ansible/modules/cloud/azure/azure_rm_devtestlab_facts.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.devtestlabs import DevTestLabsClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMLabFacts(AzureRMModuleBase):
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_devtestlab_facts.py]: AzureRMDevTestLabFacts?

File: lib/ansible/modules/cloud/azure/azure_rm_devtestlab_facts.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        self.mgmt_client = None
+        self.resource_group = None
+        self.name = None
+        self.tags = None
+        super(AzureRMLabFacts, self).__init__(self.module_arg_spec, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+        self.mgmt_client = self.get_mgmt_svc_client(DevTestLabsClient,
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_devtestlab_facts.py:180]: api version

File: lib/ansible/modules/cloud/azure/azure_rm_devtestlab_facts.py (hunk 1 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_devtestlab_facts.py:109]: "testlab6346" is a storage account name and it's generated by testlab from test lab name which is "testlab"

File: lib/ansible/modules/cloud/azure/azure_rm_devtestlab_facts.py (hunk 3 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_devtestlab_facts.py:180]: single api

**Review by yungezz — APPROVED** [review]

**Pull request merged**